mesonbuild/build.py

```diff
@@ -1062,19 +1064,24 @@ def link(self, target):
                 msg = "Can't link non-PIC static library {!r} into shared library {!r}. ".format(t.name, self.name)
                 msg += "Use the 'pic' option to static_library to build with PIC."
                 raise InvalidArguments(msg)
-            if self.is_cross != t.is_cross:
```

Comment:
@jpakkane is this just because CustomTargets don't have a `is_cross` field?

Reply:
Yes. And we can't really know either way, because the external command might have done pretty much anything.